src/proposals/20240417/SparkEthereum_20240417.t.sol

```diff
@@ -0,0 +1,183 @@
+        assertEq(aclManager.isEmergencyAdmin(FREEZER_MOM_NEW), true);
+        assertEq(aclManager.isRiskAdmin(FREEZER_MOM_NEW), true);
+    }
+
+    function testFreezerMom_REMEMBER_TO_REENABLE_AFTER_THIS_SPELL() public {
+        uint256 snapshot = vm.snapshot();
+
+        // These will run as normal
+        _runFreezerMomTests();
+
+        vm.revertTo(snapshot);
+        GovHelpers.executePayload(vm, payload, executor);
+
+        // There are new spells
+        freezerMom = ISparkLendFreezerMom(FREEZER_MOM_NEW);
+
+        // Sanity checks - cannot call Freezer Mom unless you have the hat
+        vm.expectRevert("SparkLendFreezerMom/not-authorized");
+        freezerMom.freezeMarket(Ethereum.DAI, true);
+        vm.expectRevert("SparkLendFreezerMom/not-authorized");
+        freezerMom.freezeAllMarkets(true);
+        vm.expectRevert("SparkLendFreezerMom/not-authorized");
+        freezerMom.pauseMarket(Ethereum.DAI, true);
+        vm.expectRevert("SparkLendFreezerMom/not-authorized");
+        freezerMom.pauseAllMarkets(true);
+
+        snapshot = vm.snapshot();
+
+        _assertFrozen(Ethereum.DAI,  false);
+        _assertFrozen(Ethereum.WETH, false);
+        _voteAndCast(SPELL_FREEZE_DAI);
+        _assertFrozen(Ethereum.DAI,  true);
+        _assertFrozen(Ethereum.WETH, false);
+
+        _voteAndCast(SPELL_FREEZE_ALL);
+        _assertFrozen(Ethereum.DAI,  true);
+        _assertFrozen(Ethereum.WETH, true);
+
+        _assertPaused(Ethereum.DAI,  false);
+        _assertPaused(Ethereum.WETH, false);
+        _voteAndCast(SPELL_PAUSE_DAI);
+        _assertPaused(Ethereum.DAI,  true);
+        _assertPaused(Ethereum.WETH, false);
+
+        _voteAndCast(SPELL_PAUSE_ALL);
+        _assertPaused(Ethereum.DAI,  true);
+        _assertPaused(Ethereum.WETH, true);
+
+        vm.revertTo(snapshot);
+
+        _assertFrozen(Ethereum.DAI,  false);
+        _assertFrozen(Ethereum.WETH, false);
+        vm.prank(FREEZER_MULTISIG);
+        freezerMom.freezeMarket(Ethereum.DAI, true);
+        _assertFrozen(Ethereum.DAI,  true);
+        _assertFrozen(Ethereum.WETH, false);
+
+        vm.prank(FREEZER_MULTISIG);
+        freezerMom.freezeAllMarkets(true);
+        _assertFrozen(Ethereum.DAI,  true);
+        _assertFrozen(Ethereum.WETH, true);
+
+        _assertPaused(Ethereum.DAI,  false);
+        _assertPaused(Ethereum.WETH, false);
+        vm.prank(FREEZER_MULTISIG);
+        freezerMom.pauseMarket(Ethereum.DAI, true);
+        _assertPaused(Ethereum.DAI,  true);
+        _assertPaused(Ethereum.WETH, false);
+
+        vm.prank(FREEZER_MULTISIG);
+        freezerMom.pauseAllMarkets(true);
+        _assertPaused(Ethereum.DAI,  true);
+        _assertPaused(Ethereum.WETH, true);
+
+        assertEq(freezerMom.wards(FREEZER_MULTISIG), 1);
+        _voteAndCast(SPELL_REMOVE_MULTISIG);
+        assertEq(freezerMom.wards(FREEZER_MULTISIG), 0);
+    }
+
+    function testGnosisSpellExecution() public {
```

Comment:
THis ones still dependent on gnosis payload right? We should also add a quick assertion about something small on gnosis to make sure that its actually performing the execution. expectCall here is passing with address(0)

Reply:
Ya I'm actually not sure why this is passing. Can change to something that will fail if the payload is not set properly.

Reply:
Seems like we still need to update this

Reply:
I added a check that the payload is actually set to make this fail. It's weird that delegate call succeeds even to random address, however it is still doing a valid check that the payload is not 0 and it calls execute on whatever the payload address is. I think this is sufficient.

Reply:
Resolving now that we've implemented a reserves check on gnosis